### ISSUE TITLE
Mismatch in unfolded band (supercell) with unit cell band structure

### ISSUE THREAD

**Issue by Gargee-123** [QuantumMaterialsModelling/bands4vasp#4]
Dear bands4vasp Developers, 

I have done one band unfolding of NbSe2 supercell (rectangular) on the BZ of hexagonal NbSe2 unit cell. However, I can not see clearly the unit cell band patterns in the unfolded bands. Moreover, there are many extra bands in the unfolded band structure. 

![EBSbloch](https://user-images.githubusercontent.com/88433236/141831946-d70065a6-9cb9-4d80-b708-90ef369d4c2f.png)

I have attached the inputs here with this email. Please suggest me where I have done wrong. 

[INPUTS.zip](https://github.com/QuantumMaterialsModelling/bands4vasp/files/7540866/INPUTS.zip)

**Comment by Gargee-123** [comment]
Dear Developers, 

I would like to add one more result of the same system. In the previous section, I have defined the transformation matrix in INCAR, while in this case (Figure below), I have provided the POSCAR and POSCAR.prim file without defining the transformation matrix and did the unfolding calculation. The result is attached here. However, I can not see the pattern matches with the unitcell bandstructure. Please suggest me regarding my mistake. 


![EBSbloch](https://user-images.githubusercontent.com/88433236/141836138-5063bfeb-bd50-4de7-9a58-aec9a45a4938.png)

**Comment by MicheleReticcioli** [comment]
Dear Gargee, could you upload the POSCAR.prim that you used for the the plot in the second message please? I assume this POSCAR.prim is equivalent to the POSCAR that you used to calculate the band structure of the primitive cell without unfolding, is this correct?

**Comment by Gargee-123** [comment]
Dear Michele, 

Thank you for your response. I have uploaded the folder earlier where the POSCAR.prim was generated from the POSCAR and the INTMUL. 
In the second approach (second figure), I provided the POSCAR.prim, which is slightly distorted from the original NbSe2 unit cell. Here I am attaching a folder containing the that distorted POSCAR.prim, which is same as the POSCAR (named as '**POSCAR-distorted-unit-NbSe2**', the bandstructure for this distorted NbSe2 is named as '**band-POSCAR-distorted-unit-NbSe2**'). 

This folder also contains original NbSe2 unicell (named as '**CONTCAR-unit-NbSe2**') and its bandstructure (named as '**band-CONTCAR-unit-NbSe2**').

Here I am explaining why I have chosen the distorted NbSe2 unit for unfolding. The POSCAR (supercell) in the folder is a part of misfit structure. After relaxing the complete misfit structure, I have checked and found that NbSe2 in the misfit are slightly distorted from the original NbSe2. Therefore, I have taken that NbSe2 supercell which is a part of complex misfit structure and made the unitcell of NbSe2 (which is distorted than the original NbSe2 unit cell.)

[INPUTS-2.zip](https://github.com/QuantumMaterialsModelling/bands4vasp/files/7545314/INPUTS-2.zip)


Please let me know if you have any other questions regarding this. 


Regards
Gargee

**Comment by MicheleReticcioli** [comment]
It seems there is problem with your input files. I believe there was a mistake in the order of the elements of your INTMUL option. Please refer to the wiki page 4 for a detailed explanation.
I attach here the POSCAR files that I have obtained by applying two different INTMUL matrices on your POSCAR-distorted-unit-NbSe2 file (saved here as POSCAR.prim):
POSCAR_700420001.vasp was generated using INTMUL= 7 0 0 4 2 0 0 0 1 ;
POSCAR_740020001.vasp was generated using INTMUL= 7 4 0 0 2 0 0 0 1 .
Please choose the POSCAR file that you prefer and select the appropriate INTMUL for your unfolding calculations.
[POSCARs.zip](https://github.com/QuantumMaterialsModelling/bands4vasp/files/7545622/POSCARs.zip)

**Comment by Gargee-123** [comment]
Dear Michele, 

I have checked the two POSCAR generated. None of these matches with the supercell in the misfit which I attached previously. But I think I have to give the POSCAR.prim file same as POSCAR

The distorted POSCAR file was : (where the lattice parameters are as row vector)

2 Nb Se                              
   1.0000000000000000     
     3.468   -0.014   0.028
    -1.710    2.970   0.028
     0.000    0.000  24.1608266881156375
   Nb   Se   
    1    2  
Cartesian
     0.000000000      0.000000000      12.000000000
     1.737961000      0.969301000      10.338844000
     1.740880000      0.997391000      13.678914000 



Whereas the POSCAR.prim file, I gave the lattice parameters as column vector

Automatic generated file
    1.00000000
    3.4648000000000000   -1.7100000000000000    0.0000000000000000
   -0.0140000000000000    2.9700000000000000    0.0000000000000000
    0.0280000000000000    0.0280000000000000   24.1608266881156375


I will try to do the correction and do the unfolding calculations. Thank you for your helpful suggestion.

**Comment by Gargee-123** [comment]
Dear Michele, 

I have done the unfolding with POSCAR.prim specified in the correct form and here I am attaching my result. I have added my comparison with the NbSe2 unit cell. I found that the bands are shifted up compared to the unit cell band. Can you give me any suggestion how can I fix this ?

I have also noticed a small extra part of bands (weights) in between K and M. Please also comment how to solve it. 

![EBSbloch](https://user-images.githubusercontent.com/88433236/142174029-e053f539-2825-41b5-b800-929592efcdd0.png)

[Unfolding NbSe2.pdf](https://github.com/QuantumMaterialsModelling/bands4vasp/files/7553427/Unfolding.NbSe2.pdf)


Regards
Gargee

**Comment by MicheleReticcioli** [comment]
Good that the issue is solved by using the suggested setup.
However, from the last graph it seems that you still have some broadening of the bands (low Bloch character values). I would say that this is due to a non perfect match of the ionic position of the supercell with the primitive cell (or if you have relaxed the supercell breaking the primitive-cell symmetry).
The second issue, the energy shift, could be due to the mismatch mentioned above, but also to other physical and computational quantities: have you adapted the setup of the self-consistent calculation of the supercell to the one in the primitive cell, or simply used the identical setup? For example, the k-points. 

Anyhow, glad to see that the unfolding is working for you, from the technical side. I close the issue as this is not really the place to discuss the physics of the results.
All the best with your work,
Michele

**Comment by Gargee-123** [comment]
Dear Michele, 

I relaxed the complete misfit structure and then took the NbSe2 single layer and then run a singlepoint calculation to generate the CHGCAR and WAVECAR. I did not relaxed the NbSe2 monolayer separately. The unit cell cut from that NbSe2 supercell which is used in the POSCAR.prim also. 

I have used the same parameters for self-consistent calculation. For the unit cell the K points are high, but for the supercell I have used much reduced K-points. That can be the issue. I am also running another calculation with higher K-points. 

Thanks & Regards
Gargee
